### PR TITLE
wip: replace pomegranate with alembic

### LINKED ISSUE #206: Doesn't work with Java 9
clj-refactor.el and refactor-nrepl version information

cider/cider-nrepl "0.14.0"  refactor-nrepl "2.3.1"

### CIDER version information

```
;; CIDER 0.12.0snapshot (package: 20160331.421), nREPL 0.2.12
;; Clojure 1.8.0, Java 1.8.0_31
```

### Leiningen or Boot version
Leiningen 2.8.0 on Java 9.0.1 Java HotSpot(TM) 64-Bit Server VM

### Emacs version

Emacs 25.3.1

### Operating system

MacOS 

Darwin 16.6.0 Darwin Kernel Version 16.6.0: Fri Apr 14 16:21:16 PDT 2017; root:xnu-3789.60.24~6/RELEASE_X86_64 x86_64

Error Message:
'''
$ lein repl
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by dynapath.defaults$eval380$fn__381 to method java.net.URLClassLoader.addURL(java.net.URL)
WARNING: Please consider reporting this to the maintainers of dynapath.defaults$eval380$fn__381
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Java HotSpot(TM) 64-Bit Server VM warning: Unable to open cgroup memory limit file /sys/fs/cgroup/memory/memory.limit_in_bytes (No such file or directory)
WARNING: boolean? already refers to: #'clojure.core/boolean? in namespace: mranderson047.toolsanalyzerjvm.v0v6v9.toolsanalyzer.v0v6v7.clojure.tools.analyzer.utils, being replaced by: #'mranderson047.toolsanalyzerjvm.v0v6v9.toolsanalyzer.v0v6v7.clojure.tools.analyzer.utils/boolean?
WARNING: boolean? already refers to: #'clojure.core/boolean? in namespace: mranderson047.toolsanalyzerjvm.v0v6v9.toolsanalyzer.v0v6v7.clojure.tools.analyzer, being replaced by: #'mranderson047.toolsanalyzerjvm.v0v6v9.toolsanalyzer.v0v6v7.clojure.tools.analyzer.utils/boolean?
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by mranderson047.alembic.v0v3v3.dynapath.v0v2v5.dynapath.defaults$eval18085$fn__18086 to method java.net.URLClassLoader.addURL(java.net.URL)
WARNING: Please consider reporting this to the maintainers of mranderson047.alembic.v0v3v3.dynapath.v0v2v5.dynapath.defaults$eval18085$fn__18086
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Error loading refactor-nrepl.middleware: clojure.lang.ExceptionInfo: Alembic can not manipulate specified ClassLoader. {:classloader #object[jdk.internal.loader.ClassLoaders$AppClassLoader 0x484b61fc "jdk.internal.loader.ClassLoaders$AppClassLoader@484b61fc"], :reason :not-addable-with-dynapath}, compiling:(mranderson047/alembic/v0v3v3/alembic/still.clj:65:1)
Exception in thread "main" java.lang.RuntimeException: Unable to resolve var: refactor-nrepl.middleware/wrap-refactor in this context, compiling:(/private/var/folders/bg/jpxx1rf134z369y0dmb5zt640000gn/T/form-init6665213846665153788.clj:1:8774)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6953)
	at clojure.lang.Compiler.analyze(Compiler.java:6729)
	at clojure.lang.Compiler.analyze(Compiler.java:6685)
	at clojure.lang.Compiler$InvokeExpr.parse(Compiler.java:3855)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6948)
	at clojure.lang.Compiler.analyze(Compiler.java:6729)
	at clojure.lang.Compiler.analyze(Compiler.java:6685)
	at clojure.lang.Compiler$InvokeExpr.parse(Compiler.java:3855)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6948)
	at clojure.lang.Compiler.analyze(Compiler.java:6729)
	at clojure.lang.Compiler.access$300(Compiler.java:38)
	at clojure.lang.Compiler$LetExpr$Parser.parse(Compiler.java:6324)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6946)
	at clojure.lang.Compiler.analyze(Compiler.java:6729)
	at clojure.lang.Compiler.analyze(Compiler.java:6685)
	at clojure.lang.Compiler$BodyExpr$Parser.parse(Compiler.java:6056)
	at clojure.lang.Compiler$FnMethod.parse(Compiler.java:5428)
	at clojure.lang.Compiler$FnExpr.parse(Compiler.java:3993)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6944)
	at clojure.lang.Compiler.analyze(Compiler.java:6729)
	at clojure.lang.Compiler.eval(Compiler.java:7002)
	at clojure.lang.Compiler.eval(Compiler.java:6995)
	at clojure.lang.Compiler.eval(Compiler.java:6995)
	at clojure.lang.Compiler.load(Compiler.java:7457)
	at clojure.lang.Compiler.loadFile(Compiler.java:7395)
	at clojure.main$load_script.invokeStatic(main.clj:277)
	at clojure.main$init_opt.invokeStatic(main.clj:279)
	at clojure.main$init_opt.invoke(main.clj:279)
	at clojure.main$initialize.invokeStatic(main.clj:310)
	at clojure.main$null_opt.invokeStatic(main.clj:344)
	at clojure.main$null_opt.invoke(main.clj:341)
	at clojure.main$main.invokeStatic(main.clj:423)
	at clojure.main$main.doInvoke(main.clj:386)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.lang.Var.applyTo(Var.java:702)
	at clojure.main.main(main.java:37)
Caused by: java.lang.RuntimeException: Unable to resolve var: refactor-nrepl.middleware/wrap-refactor in this context
	at clojure.lang.Util.runtimeException(Util.java:221)
	at clojure.lang.Compiler$TheVarExpr$Parser.parse(Compiler.java:716)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6946)
	... 35 more
'''

### PR DESCRIPTION
**Pull request by benalbrecht**
fixes #206

this allows hotloading in java10, but it probably requires some more testing.

### CODE REVIEW

**Comment by benedekfazekas** [comment]
wohoo! this supposed to be backward compatible too, right?

**Comment by benedekfazekas** [comment]
seems to be a genuine failure i am getting the same error as on travis locally when running `test` wo/ mranderson. I guess we need to figure out the right version of piggieback?!

File: project.clj [hunk 1]
```diff
@@ -19,21 +22,20 @@
...
                     :src-paths ["test/resources"]}
-             :1.8 {:dependencies [[org.clojure/clojure "1.8.0"]]}
              :1.9 {:dependencies [[org.clojure/clojure "1.9.0"]
                                   [org.clojure/clojurescript "1.9.908"]]}
              :dev {:plugins [[jonase/eastwood "0.2.0"]]
                    :global-vars {*warn-on-reflection* true}
-                   :dependencies [[org.clojure/clojurescript "1.9.89"]
-                                  [com.cemerick/piggieback "0.2.2"]
+                   :dependencies [[org.clojure/clojurescript "1.10.238"]
+                                  [cider/piggieback "0.2.3"]
```
Comment (bbatsov) [inline project.clj]: That should 0.3.6.

**Comment by expez** [comment]
Very exciting stuff, @benalbrecht!

File: project.clj (hunk 1 above)
Comment (benedekfazekas) [reply project.clj]: and `cemerick.piggieback` needs to be updated in `refactor-nrepl.ns.resolve-missing-test`

there are still failures and errors when tests are run but it compiles after the above changes

**Comment by bbatsov** [comment]
>  wip: replace pomegranate with alembic #224 

I think that's the other way. ;-) Btw, what's the deal with pomegranate these days? Is it maintained?

**Comment by bbatsov** [comment]
Btw, a recent patch in https://github.com/nrepl/nREPL/pull/35 might render all of this obsolete - we can probably just use `tools.deps` directly. And we can probably even move this to CIDER itself and maybe one day to nREPL itself.

**Comment by expez** [comment]
@bbatsov That sounds like a good way forward!  Good input! :+1:

**Comment by bbatsov** [comment]
@expez In light of this development I'd suggest for you to just remove the hotloading support, which will also solve the problem with getting out a new snapshot and we'll revisit this a couple of months from now when https://github.com/clojure-emacs/cider-nrepl/pull/540 is live and we're running happily on the modern nREPL.

**Comment by expez** [comment]
@bbatsov Yeah, that makes sense to me too.

**Comment by bbatsov** [comment]
I'll just mention here for completeness that the necessary change was done in nREPL (https://github.com/nrepl/nREPL/pull/35)

**Comment by bbatsov** [comment]
I think we can close this one. No need to keep open stale PRs forever.

**Pull request closed without merging**